### DISCUSSION TITLE
Variable breaks cache/ resume

### DISCUSSION THREAD

**Discussion by Lean-s** [nextflow-io/nextflow#4777] [Q&A] [answered]
Hi, 

I've a nextflow pipeline where I use a script that is located in the project directory so I used $workflow.projectDir, but when I resume the pipeline the process is not cached. Giving the full path to the script fixes the issue. I saw that there was a [bug](https://github.com/nextflow-io/nextflow/issues/2186) related to this, but that was fixed in 2021. Does anyone have an idea why this is happening, or am I missing something?

Thanks!

Here is an example, test1 gets executed while test2 is cached: 
```
process test1 {
    output:
        path("test1.txt")
    script:
    """
    cat $workflow.projectDir/test.txt > test1.txt
    """
}
process test2 {
    output:
        path("test2.txt")
    script:
    """
    cat /path/to/projectdir/test.txt > test2.txt
    """
}
workflow {
    test1()
    test2()
}
```
Nextflow version: 23.10.1

**Comment by Lean-s** [comment, marked as answer]
I found another way by using `/bin`  in the [project directory](https://www.nextflow.io/docs/latest/sharing.html#bundling-executables-in-the-workflow) and the moduleBinaries binary feature for in [modules](https://www.nextflow.io/docs/latest/module.html#module-binaries).

**Comment by bentsherman** [comment]
I think the general advice here is to never reference an input file that is not explicitly declared as a process input. That will break a number of things like containerization and caching in some cases
